packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleTextViewer/SingleTextViewerContainer.js

```diff
@@ -1,38 +1,52 @@
 import PropTypes from 'prop-types'
 import asyncStates from '@zooniverse/async-states'
 
+import { useSubjectText } from '@hooks'
+import locationValidator from '../../helpers/locationValidator'
 import SingleTextViewer from './SingleTextViewer'
 
+const defaultSubject = {
+  id: '',
+  locations: []
+}
+
 export default function SingleTextViewerContainer ({
-  content = '',
-  contentLoadingState = asyncStates.initialized,
-  error = null,
   height = '',
+  loadingState = asyncStates.initialized,
   onError = () => true,
-  onReady = () => true
+  onReady = () => true,
+  subject = defaultSubject
 }) {
-  if (contentLoadingState === asyncStates.success) {
-    onReady()
+  const { data, error, loading } = useSubjectText({
+    subject,
+    onReady,
+    onError
+})
+
+  if (loadingState === asyncStates.error) {
+    return <p>Something went wrong.</p>
```

Comment:
Would it be helpful to show `error.message` here, so that people can quote it in bug reports?

Reply:
I don’t think any other subject viewers have helpful error messages, so this isn’t essential.

Reply:
Yes, good idea.